pulp_container/app/registry_api.py

```diff
@@ -584,7 +587,29 @@ def put(self, request, path, pk=None):
         _, repository = self.get_dr_push(request, path)
 
         digest = request.query_params["digest"]
-        upload = models.Upload.objects.get(pk=pk, repository=repository)
+        try:
```

Comment:
please add some comments around what's it happening here. this will help us whenever we would look into this code in X months